fastestimator/op/numpyop/meta/one_of.py

```diff
@@ -42,6 +43,7 @@ def __init__(self, *numpy_ops: NumpyOp) -> None:
             assert outputs == op.outputs, "All ops within a OneOf must share the same outputs"
             assert self.out_list == op.out_list, "All ops within OneOf must share the same output configuration"
             assert mode == op.mode, "All ops within a OneOf must share the same mode"
+            assert ds_id == op.ds_id, "All ops within a OneOf must share the same mode"
```

Comment:
```suggestion
            assert ds_id == op.ds_id, "All ops within a OneOf must share the same ds_id"
```

Reply:
one advantage of if we could convert !ds_1 into the equivalent whitelist would be to allow logically equivalent ds_id expressions to match here, as they would for mode